src/protocols/abstract/common.js

```diff
@@ -110,6 +116,18 @@ Discarded request: ${JSON.stringify(request)}`));
     return this.state === 'ready';
   }
 
+  /**
+   * Clear pendings requests.
+   * Emits an event for each discarded pending request.
+   */
+  clear () {
+    for (const request of this._pendingRequests.values()) {
```

Comment:
why not forEach?

Reply:
I try to avoid `forEach` because it's error prone when dealing with promise. I could use it here but I prefer to stick to the same construct everywhere